### PR TITLE
Added read_condition for services and clients

### PR DESCRIPTION
**Pull request by esteve**
Connects to ros2/rclcpp#52

### CODE REVIEW

**Comment by esteve** [comment]
Linux buildfarm:

http://ci.ros2.org/job/ros2_batch_ci_linux/137/

File: rmw_opensplice_cpp/src/functions.cpp [hunk 1]
```diff
@@ -1101,6 +1078,13 @@ rmw_create_client(
     goto fail;
   }
 
+  read_condition = response_datareader->create_readcondition(
```
Comment (dirk-thomas) [inline rmw_opensplice_cpp/src/functions.cpp]: This needs to be cleaned up in the `fail` section.

Same below.

Comment (esteve) [inline rmw_opensplice_cpp/src/functions.cpp]: Fixed.

Comment (dirk-thomas) [inline rmw_opensplice_cpp/src/functions.cpp]: May be you haven't pushed the fix yet?

Comment (esteve) [inline rmw_opensplice_cpp/src/functions.cpp]: Yes, I hadn't pushed it yet. I say "Fixed" when I commit it.

**Comment by dirk-thomas** [comment]
+1

**Comment by esteve** [comment]
Buildfarm job:

http://ci.ros2.org/job/ros2_batch_ci_linux/143/

**Comment by wjwwood** [comment]
Code changes look good.

**Comment by tfoote** [comment]
+1

**Comment by jacquelinekay** [comment]
Rebased

**Comment by jacquelinekay** [comment]
+1

**Pull request merged**